rando_modules/simulate.py

```diff
@@ -20,222 +20,300 @@ def __init__(self, **kwargs):
+                if is_new_pseudoitem:
+                    has_new_pseudoitem = True
+            return has_new_pseudoitem
+
+        raise TypeError('item_object argument is not of type str or list',
+                        type(item_object),
+                        item_object)
+
+
+    def remove_from_inventory(self, item_object):
+        """Remove something from Mario's inventory."""
+        all_partners = ["Goombario", "Kooper", "Bombette", "Parakarry",
+                        "Bow", "Watt", "Sushie", "Lakilester"]
+        # Overload: Single item -> Remove item
+        if isinstance(item_object, str):
+
+            if (   item_object.startswith("GF")
+                or item_object.startswith("MF")
+                or item_object.startswith("MB")
+                or item_object.startswith("RF")
+            ):
+                self.flags.remove(item_object)
+            elif item_object in all_partners:
+                self.partners.remove(item_object)
+                self.item_history.append(f"-{item_object}")
+            elif item_object.find("StarPiece") != -1:
+                if item_object.startswith("Three"):
+                    self.starpiece_count -= 3
+                else:
+                    self.starpiece_count -= 1
+                self.starpieces.remove(item_object)
+                self.item_history.append(f"-{item_object}")
+            elif item_object.startswith("FAVOR"):
+                self.favors.remove(item_object)
+            elif item_object.startswith("EQUIPMENT"):
+                if (item_object.startswith("EQUIPMENT_Boots_Progressive")):
+                    self.boots.remove(item_object)
+                if (item_object.startswith("EQUIPMENT_Hammer_Progressive")):
+                    self.hammer.remove(item_object)
+            elif item_object.startswith("STARSPIRIT"):
+                self.starspirits.remove(item_object)
+            else:
+                self.items.remove(item_object)
+                self.item_history.append(f"-{item_object}")
+            return
+
+        # Overload: List of items -> Call function per item
+        if isinstance(item_object, list):
+            for singular_item in item_object:
+                self.remove_from_inventory(singular_item)
+            return
+
+        raise TypeError('item_object argument is not of type str or list',
+                        type(item_object),
+                        item_object)
+
+
+    def can_flip_panels(self):
+        """
+        Checks if Mario can currently flip hidden panels:
+        * Superboots or better, or
+        * UltraHammer
+        """
+        return ((len(self.hammer) - 1) >= 2 or len(self.boots) - 1 >= 1)
+
+
+    def can_shake_trees(self):
+        """
+        Checks if Mario can currently shake trees:
+        * Hammer or better, or
+        * Bombette
+        """
+        return len(self.hammer) - 1 >= 0 or "Bombette" in self.partners
```

Comment:
Is there a reason to have this instead of just len(self.hammer) >= 1?

Reply:
This is for ingame memory parity. Hammer value 0 is wooden Hammer, whereas Hammer value -1 is no Hammer. Similar with Boots, though a Boots value of -1 only disables them in battle, not on the overworld